File: Python/GnomSort.py
#GnomSort
def GnomSort(l):
    i=0
    while i<len(l)-1:
        #Check on Okay position
        if i<0 or l[i+1]>=l[i]:i+=1
        
        #Do swop if not Okay
        else:
            l[i],l[i+1]=l[i+1],l[i]
            i-=1
    return l

#Create ourList and check GnomSort
ourList=[2,6,1,8,4,5,6,3]
print(GnomSort(ourList))

File: Js/GnomSort.js
const GnomSort = (l) => {
  let sortedL = [...l];
  let i = 1;
  while (i < l.length) {
    if (i > 0 && sortedL[i - 1] > sortedL[i]) {
      [sortedL[i - 1], sortedL[i]] = [sortedL[i], sortedL[i - 1]];
      i--;
    } else {
      i++;
    }
  }
  return sortedL;
};

const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(GnomSort(l));
